### PR TITLE
Added function to remove non empty directories

### PR DESCRIPTION
**Pull request by kshalnev**

### CODE REVIEW

**Comment by syershov** [comment]
LGTM

File: platform/platform.cpp [hunk 1]
```diff
@@ -33,6 +42,38 @@ Platform::EError Platform::ErrnoToError()
...
+// static
+bool Platform::RmDirRecursively(string const & dirName)
+{
+  if (dirName.empty() || IsSpecialDirName(dirName))
+    return false;
+
+  bool res = true;
+
+  TFilesWithType fwts;
+  GetFilesByType(dirName, FILE_TYPE_REGULAR | FILE_TYPE_DIRECTORY, fwts);
```
Comment (ygorshenin) [inline platform/platform.cpp]: Probably you need to get all files in a directory, including fifos, links and other stuff.

Comment (ygorshenin) [inline platform/platform.cpp]: Just select all files by FILE_TYPE_UNKNOWN and check directories via Plaform::GetFileType()

Comment (kshalnev) [inline platform/platform.cpp]: Good point.

**Comment by kshalnev** [comment]
PTAL

File: platform/platform.cpp [hunk 2]
```diff
@@ -33,6 +42,45 @@ Platform::EError Platform::ErrnoToError()
...
+  {
+    string const path = my::JoinFoldersToPath(dirName, file);
+
+    EFileType type;
+    if (GetFileType(path, type) != ERR_OK)
+      continue;
+
+    if (type == FILE_TYPE_DIRECTORY)
+    {
+      if (!IsSpecialDirName(file))
```
Comment (ygorshenin) [inline platform/platform.cpp]:
```c++
if (!IsSpecialDirName(file) && !RmDirRecursively(path))
  res = false;
```
?

**Comment by ygorshenin** [comment]
LGTM

File: platform/platform.cpp (hunk 2 above)
Comment (kshalnev) [inline platform/platform.cpp]: Did it

**Comment by kshalnev** [comment]
@syershov @ygorshenin please merge

**Pull request merged**